app/controllers/name_controller.rb

```diff
@@ -1002,11 +1002,11 @@ def eol_data(review_status_list)
     image_data = Name.connection.select_all(%(
       SELECT name_id, image_id, observation_id, images.user_id,
              images.license_id, images.created_at
-      FROM observations, images_observations, images
+      FROM observations, observation_images, images
       WHERE observations.name_id IN (#{name_ids})
```

Comment:
Brakeman (current version) complains about this:
>

| Confidence | Class |  Method | Warning Type | Message |
| --- | --- |  --- | --- | --- |
| Medium | NameController | eol_data | SQL Injection | Possible SQL injection near line 1006: Name.connection.select_all("\n SELECT name_id, image_id, observation_id, images.user_id,\n images.license_id, images.created_at\n FROM observations, observation_images, images\n WHERE observations.name_id IN (#{{}.keys.map(&:to_s).join(",")})\n AND observations.vote_cache >= 2.4\n AND observations.id = observation_images.observation_id\n AND observation_images.image_id = images.id\n AND images.vote_cache >= 2\n AND images.ok_for_export\n ORDER BY observations.vote_cache\n ") |

It looks like a false positive to me, but I'm unsure.
It might be fixable by using Arel -- even if a false positive. The Arel is over my head.
```sql
Name.joins(observations: :images).
     where(observations: { name: name_ids }).
     where("observations.vote_cache >= 2.4").
     where("images.vote_cache >= 2").
     where(images: { ok_for_export: true})
```
will sanitize `name_ids`.
But I have no idea how to deal with the `SELECT` part of the query.

Reply:
@JoeCohen shouldnt this method really be in the model?